File: ch19p652SimplestRestorationExample/SimplestRestorationExample/DetailViewController.m


#import "DetailViewController.h"

@interface DetailViewController () // <UIViewControllerRestoration>

@end

@implementation DetailViewController


- (IBAction) unwind: (UIStoryboardSegue*) seg {
    
}

/*
 
+ (UIViewController *)viewControllerWithRestorationIdentifierPath:(NSArray *)ic
                                                            coder:(NSCoder *)coder {
    NSLog(@"%@ %@", self, ic);
    UIStoryboard* sb = [coder decodeObjectForKey:UIStateRestorationViewControllerStoryboardKey];
    return [sb instantiateViewControllerWithIdentifier:[ic lastObject]];
}


- (id)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    self.restorationClass = [self class];
    return self;
}
 
 */

-(void)encodeRestorableStateWithCoder:(NSCoder *)coder {
    NSLog(@"%@", @"dvc encode");
    [super encodeRestorableStateWithCoder:coder];
}
-(void)decodeRestorableStateWithCoder:(NSCoder *)coder {
    NSLog(@"%@", @"dvc decode");
    [super decodeRestorableStateWithCoder:coder];

}

-(void) viewDidLoad {
    [super viewDidLoad];
    NSLog(@"dvc view did load");
}

-(void)viewWillAppear:(BOOL) animated {
    [super viewWillAppear:animated];
    NSLog(@"dvc view will appear");
}

-(void)viewDidAppear:(BOOL) animated {
    [super viewDidAppear:animated];
    NSLog(@"dvc view did appear");
}

-(void)viewWillDisappear:(BOOL) animated {
    [super viewWillDisappear:animated];
    NSLog(@"dvc view will disappear");
}

-(void)viewDidDisappear:(BOOL) animated {
    [super viewDidDisappear:animated];
    NSLog(@"dvc view did disappear");
}


@end

File: bk2ch06p301customAnimation4/NavCustomPushTest/DetailViewController.swift

import UIKit

class DetailViewController : UIViewController {
    var detailItem : Any?
    @IBOutlet var iv : UIImageView!
        
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        if let im = self.detailItem as? UIImage {
            self.iv.image = im
        }
        
        print("\(self) " + #function)
        if let tc = self.transitionCoordinator {
            print(tc)
        }
    }
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        print("\(self) " + #function)
        if let tc = self.transitionCoordinator {
            print(tc)
        }
        
    }
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        print("\(self) " + #function)
        
        if let tc = self.transitionCoordinator {
            print(tc)
        }
        
    }
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        print("\(self) " + #function)
        if let tc = self.transitionCoordinator {
            print(tc)
        }
        
    }

}
